src/App.tsx

```diff
@@ -1,28 +1,71 @@
-import React, { useState } from 'react';
+import React, { SetStateAction, useState } from 'react';
 import './App.css';
 
-const Image = (props: {rotation: number}) =>
-  <img
+type State = {
+  imageIndexOffset: number
+  imagesWithRotations: { url: string, rotation: number}[]
+}
+
+function Image(props: {state: State}){
+  const currentImage = props.state.imagesWithRotations[getCurrentImageIndex(props.state)];
+  return <img
     alt="mi imagia"
-    src="https://pixabay.com/get/g1749546ed8da4018ceb48a5292b1d2607f9af1cb045c7bf6ecd9e57074795b166dd33f7d64845b8c30f00e66049bd7bc1f50d739420c29542edb16aeb55ce587_640.jpg"
-    style={{transform: `rotate(${props.rotation}deg)`}}
+    src={currentImage.url}
+    style={{transform: `rotate(${currentImage.rotation}deg)`}}
   />
+}
+
+function getCurrentImageIndex(state: State): number
+{
+  return state.imageIndexOffset % state.imagesWithRotations.length;
+}
 
-const Controls = (props: { onCounterClockwiseClick: () => void, onClockwiseClick: () => void }) =>
+function rotateCurrentImage(rotationDelta: number, prev: State): State
+{
+  const currentImageIndex = getCurrentImageIndex(prev);
+
+  //TODO: this code still has a bug in it
+  const newImages = [
```

Comment:
this works, recommend trying out with `images.map((img, index) => ...)`  to be more succinct